src/main/java/org/broadleafcommerce/menu/processor/MenuProcessorExtensionHandler.java

```diff
@@ -34,6 +34,6 @@ public interface MenuProcessorExtensionHandler extends ExtensionHandler {
      *
      * @return - ExtensionResultStatusType
      */
-    public ExtensionResultStatusType addAdditionalFieldsToModel(String tagName, Map<String, String> tagAttributes, Map<String, Object> newModelVars, BroadleafTemplateContext context);
+    public ExtensionResultStatusType addAdditionalFieldsToModel(Map<String, Object> newModelVars, String menuId, String menuName);
```

Comment:
idea of that extension is to allow some other modules override/modify result of execution of the method.
So I think signature should be something like 
public ExtensionResultStatusType addAdditionalFieldsToModel(List<MenuItemDTO> menuItems, String menuName);

or something like that... and extension will modify menuItems as needed